File: app/src/main/java/com/sagar/watchnext/network/models/tv/Show.java

package com.sagar.watchnext.network.models.tv;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class Show {

    @SerializedName("poster_path")
    private String posterPath;

    @SerializedName("popularity")
    private float popularity;

    @SerializedName("id")
    private int id;

    @SerializedName("backdrop_path")
    private String backdropPath;

    @SerializedName("vote_average")
    private float voteAverage;

    @SerializedName("overview")
    private String overview;

    @SerializedName("first_air_date")
    private String firstAirDate;

    @SerializedName("origin_country")
    private List<String> originCountries;

    @SerializedName("genre_ids")
    private List<Integer> genreIds;

    @SerializedName("original_language")
    private String originalLanguage;

    @SerializedName("vote_count")
    private int voteCount;

    @SerializedName("name")
    private String name;

    @SerializedName("original_name")
    private String originalName;

    public String getPosterPath() {
        return posterPath;
    }

    public float getPopularity() {
        return popularity;
    }

    public int getId() {
        return id;
    }

    public String getBackdropPath() {
        return backdropPath;
    }

    public float getVoteAverage() {
        return voteAverage;
    }

    public String getOverview() {
        return overview;
    }

    public String getFirstAirDate() {
        return firstAirDate;
    }

    public List<String> getOriginCountries() {
        return originCountries;
    }

    public List<Integer> getGenreIds() {
        return genreIds;
    }

    public String getOriginalLanguage() {
        return originalLanguage;
    }

    public int getVoteCount() {
        return voteCount;
    }

    public String getName() {
        return name;
    }

    public String getOriginalName() {
        return originalName;
    }
}

File: app/src/main/java/com/sagar/watchnext/network/models/tv/Show.kt
package com.sagar.watchnext.network.models.tv

import com.google.gson.annotations.SerializedName

class Show {

    @SerializedName("poster_path")
    val posterPath: String? = null

    @SerializedName("popularity")
    val popularity: Float = 0.toFloat()

    @SerializedName("id")
    val id: Int = 0

    @SerializedName("backdrop_path")
    val backdropPath: String? = null

    @SerializedName("vote_average")
    val voteAverage: Float = 0.toFloat()

    @SerializedName("overview")
    val overview: String? = null

    @SerializedName("first_air_date")
    val firstAirDate: String? = null

    @SerializedName("origin_country")
    val originCountries: List<String>? = null

    @SerializedName("genre_ids")
    val genreIds: List<Int>? = null

    @SerializedName("original_language")
    val originalLanguage: String? = null

    @SerializedName("vote_count")
    val voteCount: Int = 0

    @SerializedName("name")
    val name: String? = null

    @SerializedName("original_name")
    val originalName: String? = null
}
